### PR TITLE
[WIP] kevin hacks for perf

### PR DESCRIPTION
**Pull request by kevmoo**
Caching `isPublic` about doubles the speed of generating SDK docs

### CODE REVIEW

File: lib/src/model_utils.dart [hunk 1]
```diff
@@ -47,26 +47,42 @@ bool isInExportedLibraries(
...
-  // have a '#nodoc' in the documentation comment
-  if (e is PropertyAccessorElement && e.isSynthetic) {
-    var accessor = (e as PropertyAccessorElement);
-    if (accessor.correspondingSetter != null &&
-        !accessor.correspondingSetter.isSynthetic) {
-      e = accessor.correspondingSetter;
-    } else if (accessor.correspondingGetter != null &&
-        !accessor.correspondingGetter.isSynthetic) {
-      e = accessor.correspondingGetter;
+  var cachedVal = _privacyExpando[e];
```
Comment (rakudrama) [inline lib/src/model_utils.dart]: This lazy-init pattern can written

bool isPublic(e) => _privacyExpando[e] ??= _isPublic(e);

where `_isPublic` is the old definition of `isPublic`.

Comment (kevmoo) [inline lib/src/model_utils.dart]: Nice!

**Comment by keertip** [comment]
@kevmoo , thanks! Overlooked that dartdoc was calling `isPublic` more than once.

lgtm!

**Comment by kevmoo** [comment]
@keertip it seems that updating deps takes care of most of the perf issues – although we might reconsider this later...

**Pull request closed without merging**